src/Converter/ShipmentLineItemsConverter.php

```diff
@@ -45,19 +52,45 @@ private function convertUnitRefundToLineItem(ShipmentRefund $shipmentRefund): Li
         ;
         Assert::notNull($shippingAdjustment);
 
+        /** @var ShipmentInterface $shipment */
         $shipment = $shippingAdjustment->getShipment();
         Assert::notNull($shipment);
         Assert::isInstanceOf($shipment, AdjustableInterface::class);
         Assert::lessThanEq($shipmentRefund->total(), $shipment->getAdjustmentsTotal());
 
+        $taxAdjustment = $this->getTaxAdjustment($shipment);
+        $taxAdjustmentAmount = $taxAdjustment !== null ? $taxAdjustment->getAmount() : 0;
+
+        $grossValue = $shipmentRefund->total();
+        $taxAmount = (int) ($grossValue * $taxAdjustmentAmount / $shipment->getAdjustmentsTotal());
+        $netValue = $grossValue - $taxAmount;
+
         return new LineItem(
             $shippingAdjustment->getLabel(),
             1,
-            $shipmentRefund->total(),
-            $shipmentRefund->total(),
-            $shipmentRefund->total(),
-            $shipmentRefund->total(),
-            0
+            $netValue,
+            $grossValue,
+            $netValue,
+            $grossValue,
+            $taxAmount,
+            $this->taxRateProvider->provide($shipment)
         );
     }
+
+    private function getTaxAdjustment(ShipmentInterface $shipment): ?AdjustmentInterface
+    {
+        $taxAdjustments = $shipment->getAdjustments(AdjustmentInterface::TAX_ADJUSTMENT);
+        if ($taxAdjustments->isEmpty()) {
+            return null;
+        }
+
+        if ($taxAdjustments->count() > 1) {
+            throw MoreThanOneTaxAdjustment::occur();
```

Comment:
Should we consider the `Exception` keyword at the end of the exception name as a standard in Sylius or not?

Reply:
+1 for `...Exception` notation

Reply:
In this particular plugin, half of the exceptions have `Exception` suffix and half do not. As I remember, we wanted to avoid using this keyword but I don't insist and I don't have a good argument for either of these approaches.